ClientWeb/Evolution/Components/gene.component.ts

```diff
@@ -34,7 +35,7 @@ import {ArrayHelper} from "../Helpers/array-helper";
     inputs: ['gene'],
     //outputs: ['removeEvent:remove']
 })
-export class GeneComponent extends BaseGeneComponent implements OnInit, OnDestroy {
+export class GeneComponent extends BaseGeneComponent implements OnInit, OnDestroy, ManuallyCast<Site> {
```

Comment:
What is this - `ManuallyCast<Site> `? As I remember I modified deepCopy() to avoid this workaround from #90